### ISSUE TITLE
[Question] How to create a CSR with subject with "comma form"

### ISSUE THREAD

**Issue by Geobert** [Devolutions/picky-rs#65]
Hi,

Is there a way to create a CSR which subject is like this: "cn=common_name, c=xx" ?

When I create a `DirectoryName` like this:
```
let mut dn = DirectoryName::new_common_name("common_name");
dn.add_attr(NameAttr::CountryName, "xx");
```
My CA reads the CSR and display a subject like this: "c=xx+cn=common_name"

Which, in our system setup can be problematic :-/

**Comment by bcortier-devolutions** [comment]
Hi Geobert! Thank you for the feedback :slightly_smiling_face: 

I'm not sure but this may be related to issue https://github.com/Devolutions/picky-rs/issues/44 that is fixed in the latest beta release.
Can you try to generate a certificate using `picky = "6.0.0-beta5"` and confirm this is good for you?

If everything is good, I'll release picky 6.0.0 since numbers of issues are fixed and new API looks stable.
You can glance at the changes [here](https://github.com/Devolutions/picky-rs/blob/master/picky/CHANGELOG.md).

**Comment by Geobert** [comment]
It's better with beta5 indeed :)  The weird part is that it seems the RDN are added in the opposite order that I was expected:
```
add CN=common
add C=country
```

gives `c=country,cn=common`

but maybe it is intended and nothing that a `rev()` can't solve (I'm parsing a subject string to create a `DirectoryName`)

**Comment by bcortier-devolutions** [comment]
Could it be the system you're using that is reordering for display?
I think we're adding attributes in order (internally stored as a Vec that ends up serialized as an ASN1 SEQUENCE).
As such I would indeed expect RDN to be stored in the order you specified.
Maybe you can throw your CSR in PEM/base64 form in https://lapo.it/asn1js/ and confirm this?
I don't know if this would helps, but maybe I can implement provide an API for iterating over RDNs if you're able to parse certificate in Rust instead?

**Comment by bcortier-devolutions** [comment]
I generated a CSR with the following code:
```rust
        let mut name = DirectoryName::new();
        name.add_attr(NameAttr::CommonName, "Some Name");
        name.add_attr(NameAttr::CountryName, "Some Country");
        let csr = Csr::generate(name, &leaf_key, SignatureAlgorithm::RsaPkcs1v15(HashAlgorithm::SHA1)).unwrap();
        println!("{}", csr.to_pem().unwrap());

```

And got [this result](https://lapo.it/asn1js/#MIICcDCCAVgCAQAwKzESMBAGA1UEAwwJU29tZSBOYW1lMRUwEwYDVQQGDAxTb21lIENvdW50cnkwggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEKAoIBAQDUlqVaeM9uKK1EJVUCtY7RXgwQ7tY01q36CVhbdTLKs4Pk09iUyjlrdOh24MWHSijO2PdBo_6nQ8KwfzztV8c45F5UczJIVFxDube9BhhktiH8F__LnTgCAY8S-K6cPFDb6z674vx-cTCcF9zimg8LnKwKVis9vR986QyuG_HSfwUBsOkRkXvZ2l_Im9u1s3VYXFXtL86XN8Goq4X4HqACICgGO-tDRSmF-6EvCXMxsTKySHENTKyI9pNQ2gLsh0f4ZgF_8WFbUGoADOQmVoeg9v7RncyI8LC45wRnOPrFSh7UxMdE1Ab3B5kscbAPDg0S2ssQBoIP6nD0Go3To_jpAgMBAAGgADANBgkqhkiG9w0BAQUFAAOCAQEAbH_j0WGSOJuevt70SW1Spx2G9SQRMpzLpH-rMj4fJCx8cSfvp7ntsf5kjwGIIJG-z2p1d2D10Fs9fKegEkuPLI8HdEzp7DcPC94dU3qspsLydBT60ON00wlBDDAefggG6_AoZkL3kpCKtbV3HmAI3Zz-zc3YeQLymo-BmoNf9HukWBsfp7wPMTn79gKVPyqZ9dHeLx8Z_08LjpdysWpz_lcUGbZZp-mhKVS3-6MU_E9Imovb7_al_T7StqINv7Ar4AvEmokcX60d7TsZi4AFTz3bGH9ymVSa6ZaFik-YryC0E8rQ1vWkpb1236qkCLtVAr6CESpkeOFdTtMuf5BmeQ).

**Comment by Geobert** [comment]
You're right, the ASN.1 is correct :) Contacting the team responsible of our CA software now :) Thank you for your time!

**Comment by bcortier-devolutions** [comment]
Thank you as well for reporting :slightly_smiling_face: 
And I just published picky 6.0.0 that you may use instead.
